onnxscript/rewriter/function_rule.py

```diff
@@ -130,27 +127,17 @@ def _find_node_contains_key_in_name(
         return None
 
     def _find_node_by_type(
-        self, function: onnx.FunctionProto, domain: str, op_type: str
-    ) -> onnx.NodeProto | None:
+        self, function: ir.Function, domain: str, op_type: str
+    ) -> ir.Node | None:
         # Repeat
-        for node in function.node:
+        for node in function:
             if node.domain == domain and node.op_type == op_type:
                 return node
         return None
 
-    def _find_constant_node(
-        self, function: onnx.FunctionProto, value_name: str
-    ) -> onnx.NodeProto | None:
-        # Potentially repeat, utility function.
-        for node in function.node:
-            for output in node.output:
-                if output == value_name:
-                    return node
-        return None
-
     def compose_new_function(
-        self, old_function: onnx.FunctionProto, pkg_version: version.Version | None
-    ) -> tuple[onnx.FunctionProto, tuple[onnx.OperatorSetIdProto]]:
+        self, old_function: ir.Function, pkg_version: version.Version | None
+    ) -> tuple[ir.Function]:
```

Comment:
Is the return type accurate? I would check if mypy is enabled for this file in lintrunner.toml

Reply:
It looks like most/all of files under optimizer and rewriter is not covered in MYPY and PYLINT. Pattern-writing breaks MYPY.

Reply:
I enabled MYPY to cover this file.